library/LuaApi.cpp

```diff
@@ -3577,6 +3577,68 @@ static int internal_md5file(lua_State *L)
     }
 }
 
+/**************************
+ * Wrappers for Alias API *
+ **************************/
+
+static int internal_getAliasCommand(lua_State *L)
+{
+    const char *name = luaL_checkstring(L, 1);
+    std::string ret = Core::getInstance().GetAliasCommand(name);
+    if (!ret.empty())
+        lua_pushstring(L, ret.c_str());
+    else
+        lua_pushnil(L);
+    return 1;
+}
+
+static int internal_listAliases(lua_State *L)
+{
+    auto aliases = Core::getInstance().ListAliases();
+
+    lua_newtable(L);
+    for (const auto& p : aliases)
+    {
+        Lua::Push(L, p.first.c_str()); // alias
+        Lua::PushVector(L, p.second);
+        lua_settable(L, -3);
+    }
+    return 1;
+}
+
+static int internal_isAlias(lua_State *L)
+{
+    const char *name = luaL_checkstring(L, 1);
+    lua_pushboolean(L, Core::getInstance().IsAlias(name));
+    return 1;
+}
+
+static int internal_addAlias(lua_State *L)
+{
+    const char *new_alias = luaL_checkstring(L, 1);
+    std::vector<std::string> alias;
+    split_string(&alias, new_alias, " ");
```

Comment:
I would prefer to have this function take a list of arguments instead; otherwise there is no way to set up certain aliases, e.g. with spaces in arguments.

It looks like there is a `Lua::GetVector()` that could work here. It's a more specific function than its name would apply - it reads strings starting at the stack position you give (`1`) and pushes them all into a `vector<string>`, but that looks like it should work for our purposes here.

Reply:
I'll adjust.

Reply:
This change is complete, tests have been updated.

Reply:
I was mistaken - on second look, it looks like GetVector takes a table and converts it to a vector, but your code is expecting that.

Have you tested passing a non-table to make sure the code doesn't crash? An error message is fine (and expected).

Reply:
I have tested that, and it does crash. I'll take a look and find a way to ensure stability.

Reply:
`luaL_checktype(L, 1, LUA_TTABLE)` before the GetVector call would probably work.